spec/lib/jobs/dell-wsman-get-systemcomponents-spec.js

```diff
@@ -0,0 +1,233 @@
+// Copyright 2017, DELL EMC, Inc.
+
+'use strict';
+
+describe('Dell Wsman Get Component Config Catalog Job', function(){
+    var waterline;
+    var WsmanJob;
+    var uuid;
+    var job;
+    var sandbox = sinon.sandbox.create();
+    var configuration;
+    var WsmanTool;
+    var obms = {
+        "service" : "dell-wsman-obm-service",
+        "config" : {
+            "user" : "abc",
+            "password" : "123",
+            "host" : "1.1.1.1"
+        },
+        "node" : "59db1dc1423ad2cc0650f8bc"
+    };
+
+    var configFile = {
+        "services": {
+            "configuration": {
+                "getComponents": "/api/1.0/server/configuration/getComponents",
+            }
+        },
+        "gateway": "http://localhost:46011",
+        "shareFolder": {
+            "address": "20.20.20.20",
+            "shareName": "test",
+            "username": "adc",
+            "password": "123",
+            "shareType": 0
+        },
+    };
+    var fullResponse = {
+        "body": {
+            "status": "OK",
+            "message": "Request Success",
+            "serverComponents": [
+                {
+                    "fqdd": "LifecycleController.Embedded.1",
+                    "attributes": []
+                }
+            ]
+        }
+    };
+
+    before(function(){
+        helper.setupInjector([
+            helper.require('/spec/mocks/logger.js'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/dell-wsman-base-job.js'),
+            helper.require('/lib/jobs/dell-wsman-get-systemcomponents.js'),
+            helper.require('/lib/utils/job-utils/wsman-tool.js')
+        ]);
+        waterline = helper.injector.get('Services.Waterline');
+        WsmanJob = helper.injector.get('Job.Dell.Wsman.Get.SystemConfigurationComponents');
+        uuid = helper.injector.get('uuid');
+        configuration = helper.injector.get('Services.Configuration');
+        //BaseJob = helper.injector.get('Job.Dell.Wsman.Base');
+        WsmanTool = helper.injector.get('JobUtils.WsmanTool');
+    });
+
+    beforeEach(function(){
+        job = new WsmanJob({}, {"nodeId": uuid.v4()}, uuid.v4());
+        sandbox.stub(WsmanTool.prototype, 'clientRequest');
+        sandbox.stub(job, 'checkOBM');
+        sandbox.stub(job, 'getIpAddress');
+        waterline.nodes = {
+            findByIdentifier: sandbox.stub()
+        };
+
+        waterline.obms = {
+            findByNode: sandbox.stub()
+        };
+
+        waterline.catalogs = {
+            findLatestCatalogOfSource: sandbox.stub(),
```

Comment:
This function is originally a promise, so mock it also a promise will be better. The same for other waterline mocks.